sql: kv tracing makes distsql explode

cc @jordanlewis 

```
> CREATE TABLE group_ord (
  x INT PRIMARY KEY,
  y INT,
  z INT,
  INDEX foo(z)
);
> INSERT INTO group_ord VALUES
(1, 2, 3),
(3, 4, 5),
(5, NULL, 5),
(6, 2, 3),
(7, 2, 2),
(8, 4, 2);
> \set auto_trace on,kv
> select z, min(y) from group_ord@foo group by z;
```

panics server with:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x20cf6c6]

goroutine 439 [running]:
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/opentracing-go.SpanFromContext(0x0, 0x0, 0x18, 0xc4202c9ef0)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/opentracing-go/gocontext.go:22 +0x26
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog(0x0, 0x0, 0x2a7df1b, 0xc420721370, 0x0, 0x0)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:94 +0x39
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x0, 0x0, 0xc425580100, 0x7aa03b, 0x11, 0xc42558bbc8, 0x2, 0x2)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:109 +0x50
github.com/cockroachdb/cockroach/pkg/util/log.vEventf(0x0, 0x0, 0x1544a1221b350f00, 0x1, 0xc400000002, 0x7aa03b, 0x11, 0xc42558bbc8, 0x2, 0x2)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:178 +0x13c
github.com/cockroachdb/cockroach/pkg/util/log.VEventf(0x0, 0x0, 0x2, 0x7aa03b, 0x11, 0xc42558bbc8, 0x2, 0x2)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:193 +0x8d
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).NextRow(0xc420721370, 0x0, 0x0, 0x1d24e86, 0x5b588174, 0x23a200ca, 0x11a8f6553, 0xc42558bc70, 0x22a59c6, 0xbece3e3d23a200ca)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:982 +0x37a
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*rowFetcherWrapper).Next(0xc42033da00, 0xed2ea7874, 0x0, 0x10000, 0x0)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:137 +0x4a
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*InputStatCollector).Next(0xc42033da40, 0xca1a40, 0xc420869da0, 0xc424e04340, 0xc4208c7200)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/stats.go:52 +0x59
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*indexJoiner).Next(0xc420720c00, 0xed2ea7874, 0x0, 0x1, 0x1)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/indexjoiner.go:170 +0x7f
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*InputStatCollector).Next(0xc42033da60, 0x84edff26bf298de, 0x710fc53b155f50aa, 0x0, 0xc424daef50)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/stats.go:52 +0x59
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*orderedAggregator).accumulateRows(0xc420721800, 0x0, 0x0, 0xc424daef50, 0x0, 0xc424daef50)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/aggregator.go:496 +0x4f
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*orderedAggregator).Next(0xc420721800, 0x0, 0x0, 0x0, 0xc424daef50)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/aggregator.go:684 +0x148
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0xca1a40, 0xc420869e60, 0xca98c0, 0xc420721800, 0xc7fd40, 0xc4252d5180)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:170 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*processorBase).Run(0xc420721800, 0xca1980, 0xc420887a00, 0xc42074fa90)
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:730 +0x98
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Start
        /data/home/kena/src/go-c21-try/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:503 +0x414
```

--- a/pkg/sql/distsqlrun/indexjoiner.go
+++ b/pkg/sql/distsqlrun/indexjoiner.go
@@ -125,6 +125,7 @@ func newIndexJoiner(
 // Start is part of the RowSource interface.
 func (ij *indexJoiner) Start(ctx context.Context) context.Context {
 	ij.input.Start(ctx)
+	ij.fetcherInput.Start(ctx)
 	return ij.startInternal(ctx, indexJoinerProcName)
 }
 

